### PR TITLE
Call disconnect from a reader when the user clicks cancel during the connection flow

### LINKED ISSUE #7153: [Payments] - Tapping on cancel after "Connect to reader" does not cancel action
**Describe the bug**
When a user taps on Cancel after connecting to reader, the action is not canceled and the reader will be connected after a few seconds.

**To Reproduce**
1. Go to Manage Card Reader
2. Tap on Connect card reader
3. When reader is found, tap on Connect to reader and quickly tap cancel while it's connecting
4. Wait few seconds on the screen
5. See that after a few seconds screen is updated showing the "Connected reader" screen

**Screenshots**

https://user-images.githubusercontent.com/17252150/183858993-654fc7c5-b549-4cd3-82db-ed0b946768ef.mp4

**Expected behavior**
When cancel is clicked, it shouldn't continue to connect to the reader

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: OPPO Reno6 Z 5G
- Android version: 12
- WooCommerce Android version: 9.8-rc-1

**WordPress Environment**
<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```
</details>

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #7153 

### Description

Call disconnect from a reader when the user clicks cancel during the connection flow. It doesn't stop connection or update right away, but disconnects after the current operation is performed 

### Testing instructions

* Start connection to a reader
* Click on "cancel"
* Notice that it doesn't end up in "connected" state

### Images/gif

https://user-images.githubusercontent.com/4923871/205916211-9ac6ae75-e6d4-43f7-b512-046e80d1f734.mp4

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/connect/CardReaderConnectViewModel.kt [hunk 1]
```diff
@@ -450,6 +450,7 @@ class CardReaderConnectViewModel @Inject constructor(
 
     private fun onCancelClicked() {
         WooLog.e(WooLog.T.CARD_READER, "Connection flow interrupted by the user.")
+        launch { cardReaderManager.disconnectReader() }
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/connect/CardReaderConnectViewModel.kt:453]: ❓ We could write a test to verify that the disconnect call is being made on every cancel click. WDYT?

**Review by AnirudhBhat — APPROVED** [review]
Thanks for working on this @kidinov I've tested this and it works as expected. I've just left one minor comment about a test that is non-blocking.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/connect/CardReaderConnectViewModel.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/connect/CardReaderConnectViewModel.kt:453]: Good point! Added a test a83d5688fc6c6441f8a4239d136938dd8d9a81e8

**Comment by pachlava** [comment]
@kidinov 👋 

Could you please confirm if this PR should be added to 10.7 section of release notes, not 10.6? Thanks! (Also, there's no link to the PR in the release notes).

**Comment by kidinov** [comment]
@pachlava good catch! Yeah, the PR was merged later than expected and I forgot to update the notes. Fix is here https://github.com/woocommerce/woocommerce-android/pull/8024

**Pull request merged**